lib/tasks/vmodl_helper.rb

```diff
@@ -0,0 +1,208 @@
+# frozen_string_literal: true
+require 'active_support/core_ext/enumerable'
+require 'active_support/inflector'
+require 'optimist'
+require 'pathname'
+require 'wsdl/parser'
+require 'rbvmomi'
+require 'rbvmomi/pbm'
+require 'rbvmomi/sms'
+
+class VmodlHelper
+  class << self
+    def verify!
+      run!('verify')
+    end
+
+    def generate!
+      run!('generate')
+    end
+
+    private
+
+    def run!(mode)
+      vmodl, wsdl = options(mode).values_at(:vmodl, :wsdl)
+      new(vmodl_path: vmodl, wsdl_path: wsdl).send("#{mode}!")
+    end
+
+    def options(rake_task)
+      argv = (i = ARGV.index('--')) ? ARGV.slice((i + 1)..-1) : ARGV.dup
+      Optimist.options(argv) do
+        educate_on_error
+        opt :wsdl, 'Path to the vsphere-ws wsdl file', type: :string, required: true
+        opt :vmodl, 'Path to the vmodl.db', type: :string, default: 'vmodl.db'
+        banner <<~EOS
+          Usage:
+          rake vmodl:#{rake_task} -- --wsdl=path/to/wsdl
+        EOS
+      end
+    end
+  end
+
+  def initialize(vmodl_path:, wsdl_path:)
+    @vmodl_path = Pathname.new(vmodl_path)
+    @wsdl_path  = Pathname.new(wsdl_path)
+
+    @vmodl = load_vmodl(@vmodl_path)
+    @wsdl  = load_wsdl(@wsdl_path)
+  end
+
+  def verify!
+    # Loop through the ComplexTypes in the WSDL and compare their types
+    # to the types which are defined in the vmodl.db
+    wsdl_types_by_name.each_value do |type|
+      type_name = type.name.name
+      next if type_name.match?(/^ArrayOf/) || type_name.match(/RequestType$/)
```

Comment:
```suggestion
      next if type_name.start_with?("ArrayOf") || type_name.end_with?("RequestType")
```

Reply:
Actually this isn't needed anymore, being filtered out by https://github.com/ManageIQ/rbvmomi2/pull/9#discussion_r727459548